src/symbolize/gimli/libs_aix.rs

```diff
@@ -0,0 +1,68 @@
+use super::mystd::borrow::ToOwned;
+use super::mystd::env;
+use super::mystd::ffi::{CStr, OsStr};
+use super::mystd::io::Error;
+use super::mystd::os::unix::prelude::*;
+use super::xcoff;
+use super::{Library, LibrarySegment, Vec};
+use alloc::vec;
+use core::mem;
+
+const EXE_IMAGE_BASE: u64 = 0x100000000;
+
+/// On AIX, we use `loadquery` with `L_GETINFO` flag to query libraries mmapped.
+/// See https://www.ibm.com/docs/en/aix/7.2?topic=l-loadquery-subroutine for
+/// detailed information of `loadquery`.
+pub(super) fn native_libraries() -> Vec<Library> {
+    let mut ret = Vec::new();
+    unsafe {
+        let mut buffer = vec![mem::zeroed::<libc::ld_info>(); 64];
+        loop {
+            if libc::loadquery(
+                libc::L_GETINFO,
+                buffer.as_mut_ptr() as *mut libc::c_char,
+                (mem::size_of::<libc::ld_info>() * buffer.len()) as u32,
+            ) != -1
+            {
+                break;
+            } else {
+                if Error::last_os_error().raw_os_error().unwrap() != libc::ENOMEM {
+                    return Vec::new();
+                }
+                buffer.resize(buffer.len() * 2, mem::zeroed::<libc::ld_info>());
```

Comment:
Okay, my question re: why the control flow is like this also becomes: is there a reason this unwraps instead of using something like
```suggestion
                 match Error::last_os_error().raw_os_error() {
                     Some(libc::ENOMEM) => buffer.resize(buffer.len() * 2, mem::zeroed::<libc::ld_info>()),
                     Some(libc::EINVAL | libc::EFAULT) => /* something? */,
                     _ => unreachable!(),
                 }
```

When reasoning by negation and positive reasoning are equally available, I prefer to avoid negation.